fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexMaintenanceTest.java

```diff
@@ -405,7 +406,7 @@ void flakyMerge(boolean isGrouped,
                     assertFalse(requireFailure && i < 15, i + " merge should have failed");
```

Comment:
Have you run this and `LuceneIndexTest` with `LUCENE_USE_LEGACY_ASYNC_TO_SYNC` defaulting to `false`, so that all the code uses the new path?

Reply:
Initially planned to add tests with the second PR, but this is a good idea. Done.